pkg/flow/internal/controller/node_native_component.go

```diff
@@ -90,7 +90,7 @@ type ComponentNode struct {
 	registry          *prometheus.Registry
 	exportsType       reflect.Type
 	moduleController  ModuleController
-	OnComponentUpdate func(cn *ComponentNode) // Informs controller that we need to reevaluate
+	OnComponentUpdate func(cn NodeWithDependants) // Informs controller that we need to reevaluate
```

Comment:
It feels off to me that we're not passing the ComponentNode interface to a function called OnComponentUpdate. Can we change the function name or the type passed to the function so they align?